File: C++/Quick Sort.cpp
#include <bits/stdc++.h>

using namespace std;

int partition(int *arr, int start, int end)
{
    int pivot = arr[end];
    int startIndex = start;
    for (int i = start; i <= end - 1; i++)
    {
        if (arr[i] <= pivot)
        {
            swap(arr[i], arr[startIndex]);
            startIndex = startIndex + 1;
        }
    }
    swap(arr[startIndex], arr[end]);
    return startIndex;
}
void quicksort(int *arr, int start, int end)
{
    int mid;
    if (start < end)
    {
        mid = partition(arr, start, end);
        quicksort(arr, start, mid - 1);
        quicksort(arr, mid + 1, end);
    }
}
int main()
{
    int size;
    cin >> size;
    int *arr = new int[size];
    for (int i = 0; i < size; i++)
    {
        cin >> arr[i];
    }
    quicksort(arr, 0, size - 1);
    cout << "\n";
    for (int i = 0; i < size; i++)
    {
        cout << arr[i];
    }
}

File: Java/Quicksort.java
import java.util.Arrays;
import java.util.Scanner;

public class Quicksort {

    /**
     * Java method to partition array by placing elements lower than pivot to left
     * of pivot and those higher than pivot to the right
     * 
     * @param int[] arr : array to be sorted
     * @param int   low : starting index
     * @param int   high : ending index
     * @return index of pivot
     */
    public static int partition(int[] arr, int low, int high) {
        int pivot = arr[high];
        int i = low - 1;
        for (int j = low; j < high; j++) {
            if (arr[j] < pivot) {
                i++;
                int temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
        }
        arr[high] = arr[i + 1];
        arr[i + 1] = pivot;
        return i + 1;
    }

    /**
     * Java method to sort array using quicksort
     * 
     * @param int[] arr : array to be sorted
     * @param int   low : starting index
     * @param int   high : ending index
     */
    public static void quicksort(int[] arr, int low, int high) {
        if (low < high) {
            int pi = partition(arr, low, high);
            quicksort(arr, low, pi - 1);
            quicksort(arr, pi + 1, high);
        }
    }

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int n;
        System.out.println("Enter the number of array elements: ");
        n = sc.nextInt();

        int arr[] = new int[n];
        System.out.println("Enter the array elements: ");
        for (int i = 0; i < n; i++)
            arr[i] = sc.nextInt();

        System.out.println("Original array : " + Arrays.toString(arr));
        quicksort(arr, 0, arr.length - 1);
        System.out.println("Sorted array : " + Arrays.toString(arr));

        sc.close();
    }
}
